### ISSUE TITLE
2024-08a break dns resolution

### ISSUE THREAD

**Issue by osos** [mailcow/mailcow-dockerized#6042]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... I understand that not following the below instructions will result in immediate closure and/or deletion of my issue.
- [X] ... I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X] ... I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] ... I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

### Description

```plain text
After sudo ./update.sh unbound failed with "unable to ping 1.1.1.1 and so on.

Works fine outside docker.

Now nothing can get access to internet due to
```

### Logs:

```plain text
unbound-mailcow-1    | 2024-08-21T04:53:22.654519511Z 2024-08-21 06:53:22: Healthcheck: DNS Resolution Failed on attempt 3 for hub.docker.com! Trying again...
unbound-mailcow-1    | 2024-08-21T04:53:22.659959708Z 2024-08-21 06:53:22: Healthcheck: DNS Resolution not possible after 3 attempts for hub.docker.com... Gave up!
unbound-mailcow-1    | 2024-08-21T04:53:22.665430854Z 2024-08-21 06:53:22: Healthcheck: Too many DNS failures (1 failures allowed, you got 3 failures), marking Healthcheck as unhealthy...

postfix-mailcow-1    | 2024-08-21T04:53:39.359919355Z Aug 21 06:53:39 1e2aa1500bac postfix/dnsblog[444]: warning: dnsblog_query: lookup error for DNS query 27.162.32.193.bl.spameatingmonkey.net: Host or domain name not found. Name service error for name=27.162.32.193.bl.spameatingmonkey.net type=A: Host not found, try again
postfix-mailcow-1    | 2024-08-21T04:53:39.359967340Z Aug 21 06:53:39 1e2aa1500bac postfix/dnsblog[447]: warning: dnsblog_query: lookup error for DNS query 27.162.32.193.bl.ipv6.spameatingmonkey.net: Host or domain name not found. Name service error for name=27.162.32.193.bl.ipv6.spameatingmonkey.net type=A: Host not found, try again
postfix-mailcow-1    | 2024-08-21T04:53:39.359985917Z Aug 21 06:53:39 1e2aa1500bac postfix/dnsblog[448]: warning: dnsblog_query: lookup error for DNS query 27.162.32.193.hostkarma.junkemailfilter.com: Host or domain name not found. Name service error for name=27.162.32.193.hostkarma.junkemailfilter.com type=A: Host not found, try again
postfix-mailcow-1    | 2024-08-21T04:53:39.360633235Z Aug 21 06:53:39 1e2aa1500bac postfix/dnsblog[443]: warning: dnsblog_query: lookup error for DNS query 27.162.32.193.bl.suomispam.net: Host or domain name not found. Name service error for name=27.162.32.193.bl.suomispam.net type=A: Host not found, try again

clamd-mailcow-1      | 2024-08-21T04:53:04.640824292Z ERROR: Wed Aug 21 04:53:04 2024 -> remote_cvdhead: Download failed (6) ERROR: Wed Aug 21 04:53:04 2024 ->  Message: Couldn't resolve host name
clamd-mailcow-1      | 2024-08-21T04:53:04.640977610Z WARNING: Wed Aug 21 04:53:04 2024 -> Failed to get daily database version information from server: https://database.clamav.net
clamd-mailcow-1      | 2024-08-21T04:53:04.641028450Z ERROR: Wed Aug 21 04:53:04 2024 -> check_for_new_database_version: Failed to find daily database using server https://database.clamav.net.
clamd-mailcow-1      | 2024-08-21T04:53:04.641006999Z Wed Aug 21 04:53:04 2024 -> Giving up on https://database.clamav.net...
clamd-mailcow-1      | 2024-08-21T04:53:04.641221555Z ERROR: Wed Aug 21 04:53:04 2024 -> Update failed for database: daily
clamd-mailcow-1      | 2024-08-21T04:53:04.641274624Z ERROR: Wed Aug 21 04:53:04 2024 -> Database update process failed: HTTP GET failed
clamd-mailcow-1      | 2024-08-21T04:53:04.641376022Z ERROR: Wed Aug 21 04:53:04 2024 -> Update failed.
```

### Steps to reproduce:

```plain text
1. sudo ./update.sh
```

### Which branch are you using?

master

### Which architecture are you using?

x86

### Operating System:

Ubuntu 22.04

### Server/VM specifications:

Xeon

### Is Apparmor, SELinux or similar active?

no

### Virtualization technology:

docker on raw metal

### Docker version:

27.1.2

### docker-compose version or docker compose version:

v2.29.1

### mailcow version:

2024-08a

### Reverse proxy:

nginx

### Logs of git diff:

```plain text
*
```

### Logs of iptables -L -vn:

```plain text
*
```

### Logs of ip6tables -L -vn:

```plain text
*
```

### Logs of iptables -L -vn -t nat:

```plain text
*
```

### Logs of ip6tables -L -vn -t nat:

```plain text
*
```

### DNS check:

```plain text
Error response from daemon: No such container: dig
```

**Comment by osos** [comment]
Turns out it was some container hanging during the sudo ./update.sh command shutting down the existing mailcow containers.

Some container (didn't find it) was still hanging on the network bridge. For some reason blocking unbound to establish internet connection or at least no dns resolution.

Solution was to
1) stop the mailcow containers with `docker compose stop`. 
2) restart dokcer with `service docker restart `
3) prune the system with `docker system prune -a`
4) `docker compose up -d`

**Comment by jinzo** [comment]
Just another point, I was bitten by this too. So it is not a completely one-off.

**Comment by apio-sys** [comment]
Not noted this and I'm always running latest stable. Have you tried the solution proposed by @osos above? This does seem an exception IMHO.

**Comment by jinzo** [comment]
Yes, the solution proposed above worked. This is my first issue with Mailcow (and been running it for quite some time too) - but still thought that reporting that he/they are not the only ones affected by this. I updated I think on Monday? And had delivery and sending problem(s) until I followed the solution instructions.

**Comment by apio-sys** [comment]
Good that it solved your issue. I think we can still consider this closed as if it would have been a general issue, we would have seen a lot of those reports. I haven't seen it. Maybe as best practice, before any updating is done, make sure your OS is stable, i.e. no pending updates, no pending boot required and then run your update (preferably from a freshly booted system after a few minutes of warm-up and confirming all the stack is working before starting the update.

**Comment by DerLinkman** [comment]
We get these messages here and there. But mostly it is a host issue.

If you setup a mailcow server even on standardized nat settings without LAN restrictions outbound, it will work flawlessly. 

Unbound simply runs a script in background with ping and dig/dns checks. That's all! 

I don't know and i don't have enough time to "debug" the issues of a few people while the most users have no complaint regarding this, im sorry...

**Comment by osos** [comment]
I noticed that while running the update.sh script that the unbound container failed to stop. However, the script continued running and performing the rest of the update. And tried to bring up a new unbound container. 

I believe that something happened with it which put it into unknown state.

Maybe the script should be amended to simply test that al containers successfully stopped before continuing.

Basically it seemed my situation was that I lacked the skills to identify and kill the hanging container - believed to be the unbound one.

It is a troublesome situation to need to reboot the server, as al other services and docker containers will be taken down during that process.

**Comment by akmal** [comment]
Well, The above solution didn't fix it for me.

743497621ca2:/# ./healthcheck.sh
2024-08-29 21:40:34: Healthcheck: DNS Resolution Failed on attempt 1 for fuzzy.mailcow.email! Trying again...
2024-08-29 21:40:36: Healthcheck: DNS Resolution Failed on attempt 2 for fuzzy.mailcow.email! Trying again...
2024-08-29 21:40:38: Healthcheck: DNS Resolution Failed on attempt 3 for fuzzy.mailcow.email! Trying again...
2024-08-29 21:40:38: Healthcheck: DNS Resolution not possible after 3 attempts for fuzzy.mailcow.email... Gave up!
2024-08-29 21:40:40: Healthcheck: DNS Resolution Failed on attempt 1 for github.com! Trying again...
2024-08-29 21:40:42: Healthcheck: DNS Resolution Failed on attempt 2 for github.com! Trying again...
2024-08-29 21:40:44: Healthcheck: DNS Resolution Failed on attempt 3 for github.com! Trying again...
2024-08-29 21:40:44: Healthcheck: DNS Resolution not possible after 3 attempts for github.com... Gave up!
2024-08-29 21:40:46: Healthcheck: DNS Resolution Failed on attempt 1 for hub.docker.com! Trying again...
2024-08-29 21:40:48: Healthcheck: DNS Resolution Failed on attempt 2 for hub.docker.com! Trying again...
2024-08-29 21:40:50: Healthcheck: DNS Resolution Failed on attempt 3 for hub.docker.com! Trying again...
2024-08-29 21:40:50: Healthcheck: DNS Resolution not possible after 3 attempts for hub.docker.com... Gave up!
2024-08-29 21:40:50: Healthcheck: Too many DNS failures (1 failures allowed, you got 3 failures), marking Healthcheck as unhealthy...


743497621ca2:/# nslookup github.com
Server:		127.0.0.11
Address:	127.0.0.11#53

Non-authoritative answer:
Name:	github.com
Address: 140.82.116.4

**Comment by akmal** [comment]
I think it is premature to close this issue

**Comment by DerLinkman** [comment]
> 743497621ca2:/# nslookup github.com Server: 127.0.0.11 Address: 127.0.0.11#53
> 
> Non-authoritative answer: Name: github.com Address: 140.82.116.4

That usage is wrong.

You need to run DNS Checks against 127.0.0.1 instead. Remember unbound is the DNS recursor here, your check checks it against the host resolver which is in docker the ip 127.0.0.11

You need to trace the dns request unbound is trying in your network to see who is blocking his dns recursions.

**Comment by pferreir** [comment]
I bumped into this issue today. @osos's solution worked. Thanks a lot!

**Comment by andyraddatz** [comment]
I got this issue too and osos's solution did not work.  Healthcheck fails but doing `docker exec -it [my-ubound-node-id] bash` and i can do nslookup just fine.  is ubound config pointing to the right DNS server?

`docker logs [my-ubound-node-id]`

```
2025-02-02 13:11:15: Healthcheck: DNS Resolution Failed on attempt 1 for fuzzy.mailcow.email! Trying again...
2025-02-02 13:11:17: Healthcheck: DNS Resolution Failed on attempt 2 for fuzzy.mailcow.email! Trying again...
2025-02-02 13:11:19: Healthcheck: DNS Resolution Failed on attempt 3 for fuzzy.mailcow.email! Trying again...
2025-02-02 13:11:19: Healthcheck: DNS Resolution not possible after 3 attempts for fuzzy.mailcow.email... Gave up!
2025-02-02 13:11:21: Healthcheck: DNS Resolution Failed on attempt 1 for github.com! Trying again...
2025-02-02 13:11:23: Healthcheck: DNS Resolution Failed on attempt 2 for github.com! Trying again...
2025-02-02 13:11:26: Healthcheck: DNS Resolution Failed on attempt 3 for github.com! Trying again...
2025-02-02 13:11:26: Healthcheck: DNS Resolution not possible after 3 attempts for github.com... Gave up!
2025-02-02 13:11:28: Healthcheck: DNS Resolution Failed on attempt 1 for hub.docker.com! Trying again...
2025-02-02 13:11:30: Healthcheck: DNS Resolution Failed on attempt 2 for hub.docker.com! Trying again...
2025-02-02 13:11:32: Healthcheck: DNS Resolution Failed on attempt 3 for hub.docker.com! Trying again...
2025-02-02 13:11:32: Healthcheck: DNS Resolution not possible after 3 attempts for hub.docker.com... Gave up!
2025-02-02 13:11:32: Healthcheck: Too many DNS failures (1 failures allowed, you got 3 failures), marking Healthcheck as unhealthy...
```

`docker exec -it [my-ubound-node-id] bash`

```
5850c6d90f21:/# nslookup hub.docker.com
Server:		127.0.0.11
Address:	127.0.0.11#53

Non-authoritative answer:
hub.docker.com	canonical name = elb-default.us-east-1.aws.dckr.io.
Name:	elb-default.us-east-1.aws.dckr.io
Address: 100.29.167.200
Name:	elb-default.us-east-1.aws.dckr.io
Address: 54.156.14.194
Name:	elb-default.us-east-1.aws.dckr.io
Address: 54.208.73.48
Name:	elb-default.us-east-1.aws.dckr.io
Address: 2600:1f18:2148:bc02:9528:bc49:346f:402c
Name:	elb-default.us-east-1.aws.dckr.io
Address: 2600:1f18:2148:bc00:d52:7a58:6696:6b41
Name:	elb-default.us-east-1.aws.dckr.io
Address: 2600:1f18:2148:bc01:f598:4f18:fc3a:1e2d
```

UPDATE: `healthcheck.sh` inside the container has 127.0.0.1 hardcoded, does this need to be?

```
...
for domain in "${domains[@]}" ; do
    success=false
    for ((i=1; i<=3; i++)); do
        dig_output=$(dig +short +timeout=2 +tries=1 "$domain" @127.0.0.1 2>/dev/null)
        dig_rc=$?
...
```

**Comment by alangrafu** [comment]
I can reproduce this issue with 2025-02

**Comment by Technerd-SG** [comment]
I have the same issue.

**Comment by Technerd-SG** [comment]
Update:
My problem was that my rooter was using AdguardHome (overwrite function) and when I removed the custom DNS it magically worked.
(I never had a problem with my dns configs bevore). Look at your network and dns configs maybe that solves your problem!

**Comment by Ajexsen** [comment]
The solution suggested by @osos didn't fix the issue for me. Instead, I had to modify the Unbound configuration (`./data/conf/unbound/unbound.conf` under `server:` block) by adding the following forwarding setup:
```
forward-zone:
  name: "."
  forward-addr: 1.1.1.1
  forward-addr: 8.8.8.8
```
I'm not sure if this is the ideal or correct fix, but it worked in my case. All services are running properly, and I can send and receive emails without any issues again. (will try to look for a better fix, but at least I have my email again)

**Comment by patschi** [comment]
> I'm not sure if this is the ideal or correct fix, but it worked in my case.

You might get issues when using public resolvers - as per docs:
```text
Please do not use a public resolver like we did in the example above. Many - if not all - blacklist lookups will fail with public resolvers, because blacklist server has limits on how much requests can be done from one IP and public resolvers usually reach this limits.
Important: Only DNSSEC validating DNS services will work.
```
https://docs.mailcow.email/manual-guides/Unbound/u_e-unbound-fwd/

So not the best idea. But if Cloudflare and Google DNS works for you, I'd assume something's blocking your access to the DNS root nameserver.

**Comment by Ajexsen** [comment]
> So not the best idea. But if Cloudflare and Google DNS works for you, I'd assume something's blocking your access to the DNS root nameserver.

@patschi Thanks for the tip! I was just about to check that. I've reverted the configuration changes and restarted the Unbound container. I also verified connectivity to all root servers listed in /etc/unbound/root.hints — everything is working as expected.

To monitor the behavior more closely, I created a copy of the healthcheck script and added logging for successful DNS lookups. The healthcheck now passes consistently most of the time. However, I've noticed that roughly every 10th check fails to resolve `github.com` and `hub.docker.com` (while `fuzzy.mailcow.email` continues to resolve successfully). I'm still looking into the root cause of this intermittent issue.

At least the temporary change to the Unbound configuration, using public resolvers, helped me complete the upgrade and bring the service back to a healthy state. It's now running again with the original configuration. Will keep an eye out if anything breaks.
